src/client/components/Comment/Comment.js

```diff
@@ -2,261 +2,308 @@
+      <Col md={8}>
+        <span className="comment-header-text">
+          <strong>{comment.author.fullName}</strong> commented {moment(comment.created).fromNow()}
+        </span>
+      </Col>
+      <Col md={4} mdOffset={0}>
+        {showButtons &&
+          <ButtonToolbar
+            className="comment-header-buttons pull-right"
+          >
+            <Button
+              onClick={onCommentEdit}
+              bsSize={buttonSize}
+            >
+              <Glyphicon glyph="edit" />
+            </Button>
+            <Button
+              bsSize={buttonSize}
+            >
+              <Glyphicon glyph="remove" />
+            </Button>
+          </ButtonToolbar>
+        }
+      </Col>
+    </Row>
+  )
+}
+
+function CommentBody(props) {
+  if (props.markdown) {
+    return (
+      <span
+        className="comment-body-markdown"
+        dangerouslySetInnerHTML={props.children}
+      />)
+  }
+  if (props.onoStyle) {
+    return (
+      <span
+        className="comment-body-onostyle"
+        dangerouslySetInnerHTML={props.children}
+      />
+    )
+  }
+  return (
+    <span className="comment-body-onostyle">
+      {props.children}
+    </span>
+  )
 }
 
 class Comment extends Component {
   /* eslint-disable react/sort-comp */
 
   constructor(props: Props) {
     super(props)
+    marked.setOptions({
+      gfm: true,
+      tables: true,
+      breaks: true,
+      pedantic: false,
+      sanitize: true,
+      smartLists: true,
+      smartypants: false,
+    })
+    const rendered = this.renderCommentText(props.comment.text, props)
     this.state = {
       editMode: props.newComment,
       newComment: props.newComment,
-      commentText: '',
+      commentText: props.comment.text,
+      renderedText: rendered,
     }
   }
 
   props: Props
 
+  formatOnoText(text) {
+    let formatted = escapeHTML(text)
+    formatted = formatted.replace(/@([a-z]+)/g, '<b>@$1</b>')
+    formatted = linkifyRevisionHashes(formatted)
+    // TODO add urlification of urls, revision hashes, and cases
+    return formatted
+  }
+
+  formatMarkdown(text) {
+    let formatted = marked(text)
+    formatted = linkifyRevisionHashes(formatted)
+    return formatted
+  }
+
   onCommentEdit = () => {
-    this.setState({ editMode: true })
+    this.setState({ editMode: !this.state.editMode })
+  }
+
+  onCommentDelete = () => {
+    this.props.onCommentDelete()
   }
 
   onCommentCancel = () => {
+    if (this.state.newComment) {
+      this.props.onCommentCancel()
+    }
     this.setState({
       editMode: false,
       newComment: false,
+      commentText: this.props.comment.text,
     })
-    // TODO: unmount component if this is a new comment.
   }
-  // onCommentDelete() {
-  //   // TODO: dispatch reducer action to delete comment
-  // }
 
   onCommentSave = () => {
     this.setState({
       editMode: false,
       newComment: false,
+      renderedText: this.renderCommentText(this.state.commentText, this.props),
```

Comment:
We don't need to keep this in state, pass comment text to `CommentBody`, let it operates with the rendering and transforming of the text. Also in the CommentBody you have necessary parameters, you are doing check in on `markdown` or `onoStyle` twice, when you call `renderCommentText` here and when call `render` of  `CommentBody`.